src/plugins/telemetry_collection_manager/server/telemetry_saved_objects_client.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectsFindOptions, SavedObjectsFindResponse } from 'src/core/server';
+import { SavedObjectsClient } from '../../../core/server';
+
+/**
+ * Extends the SavedObjectsClient to fit the telemetry fetching requirements (i.e.: find objects from all namespaces by default)
+ */
+export class TelemetrySavedObjectsClient extends SavedObjectsClient {
+  /**
+   * Find the SavedObjects matching the search query in all the Spaces by default
+   * @param options
+   */
+  async find<T = unknown>(options: SavedObjectsFindOptions): Promise<SavedObjectsFindResponse<T>> {
```

Comment:
I understand the need to create a new `find` method for telemetry needs but am worried that if every service/plugin creates their own override, we'll end up with a mess.

Would it not be better to enhance the `SavedObjectsClient` itself rather than implement our own home-rolled version?

Reply:
To fully understand this comment: are you suggesting we default to `namespaces:['*']` in the base `SavedObjectsClient` instead of creating a Telemetry-specific wrapper?

Reply:
Nope, I'm suggesting we have an enhanced `SavedObjectsClient` `find` method (maybe `findFromAllSpaces`(name TBD)), that everyone can use without home-rolling their own.